apps/reaper/lib/reaper/decoder/json.ex

```diff
@@ -5,6 +5,18 @@ defmodule Reaper.Decoder.Json do
   @behaviour Reaper.Decoder
 
   @impl Reaper.Decoder
+
+  def decode({:file, filename}, %{technical: %{topLevelSelector: top_level_selector}} = _dataset)
+      when not is_nil(top_level_selector) do
+    with {:ok, query} <- Jaxon.Path.parse(top_level_selector),
+         {:ok, data} <- json_file_query(filename, query) do
+      {:ok, data}
+    else
+      {:error, error} ->
+        {:error, File.read!(filename), error}
```

Comment:
Do we really want to log out the whole file?

Reply:
This maintains the existing functionality. If we don't want to do this that's another discussion I feel.